src/ast/ast_validations.c

```diff
@@ -1410,6 +1410,11 @@ static AST_Validation _ValidateUnion_Clauses(const AST *ast) {
 	// Require all RETURN clauses to perform the exact same projection.
 	uint *return_indices = AST_GetClauseIndices(ast, CYPHER_AST_RETURN);
 	uint return_clause_count = array_len(return_indices);
+	// We should have one more RETURN clause than we have UNION clauses.
+	if(return_clause_count != union_clause_count + 1) {
+			QueryCtx_SetError("Found %d UNION clauses but only %d RETURN clauses.", union_clause_count, return_clause_count);
```

Comment:
See if it makes sense to utilize our newly added errors.h here or in other places in this PR.

Reply:
My thinking for `errors.h` so far has been to only utilize it for any error message that appears more than once. That would cause us to move the "Expected boolean predicate" error there, but no other - sound sensible?